### ISSUE TITLE
Archive Dojo Users

### ISSUE THREAD

**Issue by DanielBrierton** [CoderDojo/community-platform#1177]
I was talking to someone at a recent mentor meetup and they suggested this feature:

The idea is to be able to archive your previous user relationships to dojos on demand. The benefit of this is that allows Dojos to have a manage users interface for only their currently active users, while still retaining stats and the ability to contact users.

* Users should get an email to unarchive themselves for next term (should create a new usersdojo)
* Users can rejoin a Dojo through the Dojo listing (new usersdojo)
** Some message should be displayed to returning users to explain why they need to rejoin
* When archiving the user should be able to name the archive group (e.g. "Class of 2016")
* Display/search of individual archives would be nice
* Manual archive of individual users should allow you to select a specific archive (latest default selected) would be nice to have. An email probably shouldn't be sent for this, since it's a specific action against a specific user.

@rosalanghammer Any thoughts/comments on this? @Wardormeur and I think it's a great idea and would be very valuable for dojo admins.

**Comment by rosalanghammer** [comment]
Think this is a great idea, essentially inactive users. 

All seems perfect in spec but 'Users should get an email to unarchive themselves for next term (should create a new usersdojo)' - are you thinking when they are 'archived' this email goes? 

Also needs work for any email functionality where active users are default and you only use inactive users by choice (e.g. notify users).

**Comment by Wardormeur** [comment]
>Also needs work for any email functionality where active users are default and you only use inactive users by choice (e.g. notify users).

Do you mean to ensure that every emailing function from dojo perspective must filter on the current term? That's a valid concern, but not only for emails but any dojo user listing

> are you thinking when they are 'archived' this email goes
Yes, basically; when the user is archived by a dojoadmin, he'll receive an email inviting him to continue for next term. Whatever his response, he has been archived on the previous term

**Comment by rosalanghammer** [comment]
Yes every emailing function (within a Dojo) should filter based only on active users, also features like filter should only work within a specific section. 

Thoughts could be this could be an additional panel to 'Active', and 'Pending' but we need to look at the full UI. 

Yes we can work on this 'archiving notification' when this is being built out. I think it should be a one time email that users can go back on at any point as opposed to every 'term'.
